cmd/infra/azure/create.go

```diff
@@ -148,13 +160,31 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 	}
 	l.Info("Successfully assigned contributor role to managed identity", "name", identityID)
 
-	// Create network security group
-	securityGroupName, securityGroupID, err := createSecurityGroup(ctx, subscriptionID, resourceGroupName, o.Name, o.InfraID, o.Location, azureCreds)
-	if err != nil {
-		return nil, err
+	securityGroupID := ""
+
+	if o.NetworkSecurityGroup != "" {
+		//check that the provided network security group exists
+		securityGroupClient, err := armnetwork.NewSecurityGroupsClient(subscriptionID, azureCreds, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create security group client: %w", err)
+		}
+		nsg, err := securityGroupClient.Get(ctx, resourceGroupName, o.NetworkSecurityGroup, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get network security group: %w", err)
+		}
+		securityGroupID = *nsg.ID
+		result.SecurityGroupName = o.NetworkSecurityGroup
+		l.Info("Successfully found existing network security group", "name", o.NetworkSecurityGroup)
+	} else {
+		// Create network security group
+		securityGroupName, nsgID, err := createSecurityGroup(ctx, subscriptionID, resourceGroupName, o.Name, o.InfraID, o.Location, azureCreds)
+		if err != nil {
+			return nil, err
+		}
+		securityGroupID = nsgID
+		result.SecurityGroupName = securityGroupName
+		l.Info("Successfully created network security group", "name", securityGroupName)
 	}
-	result.SecurityGroupName = securityGroupName
-	l.Info("Successfully created network security group", "name", securityGroupName)
```

Comment:
You could just print out `result.SecurityGroupName` here and remove the print outs from the if/else.

Reply:
That would leave you with a misleading message. If you specifed the security group you want to use it would then tell you that it has created a security group.